### PR TITLE
Add full file context menu to file tree / list

### LINKED ISSUE #48: Improve the file list's context menu
The context menu in the right-side file-list (staging and commit view) should contain all options that exist in the context menu of the diff view's file header

image

### PR DESCRIPTION
**Pull request by exactly-one-kas**
grafik

Fixes #48, relates to #92

@Murmele should we put this into 1.2.0?

### CODE REVIEW

**Comment by Murmele** [comment]
@exactly-one-kas from my side we can. Currently we did not fix a release date, and I think this is quite cool. Thanks!

**Comment by Murmele** [comment]
- when I select a folder and do ignore. It crashes

**Comment by Murmele** [comment]
I think it is worth writing some unittests for it. If I find the time I can start today with them

Unittests:
- ignore single file shall not ignore complete folder (and shall not crash)
- ignoring folder
- discard file shall not discard complete folder
- discard folder shall not discard other folders/files

**Comment by Murmele** [comment]
we can remove than `void TreeView::discard(const QModelIndex &index, const bool force)` because it is not used anymore

**Comment by Murmele** [comment]
Old:
`TreeView::discard(QModelIndex, bool) -> TreeView::discard(DiffTreeModel, QModelIndex) -> DiffTreeModel->discard(QModelIndex)`

In DiffTreeModel we find all files if the index is a folder. We need something like this also. Shall we check in the FileContextMenu that "files" is not empty? Because if it is empty we are discarding everything which is a catastrophic fault. 

Is it possible to show the list by default?
image

**Comment by exactly-one-kas** [comment]
Yes, that would be even better, the user wouldn't lose files that haven't been detected yet or are in the other TreeView

**Comment by Murmele** [comment]
> Yes, that would be even better, the user wouldn't lose files that haven't been detected yet or are in the other TreeView

Exactly. So the test is correctly written then ^^. For the partially staged files the user has too care then

**Comment by Murmele** [comment]
@exactly-one-kas would you like to finish this one for V1.2.0 or should we move it to the next version?

**Comment by exactly-one-kas** [comment]
Uhh, I'm not sure what I'm supposed to do here - I only see that the tests are failing? Should I take a look at these?

**Comment by Murmele** [comment]
> 

Yes I wrote some additional tests for the discard, because this is a quite critical. But it seems something is not implemented completely

File: test/TreeView.cpp [hunk 1]
```diff
@@ -86,6 +88,100 @@ void TestTreeView::restoreStagedFileAfterCommit() {
...
+    stagedTree->selectionModel()->clearSelection();
+    stagedTree->selectionModel()->select(folder1, QItemSelectionModel::Select);
+  }
+
+  DoubleTreeWidget::showFileContextMenu(QPoint(), repoView, stagedTree, true);
+
+  auto *menu = doubleTree->findChild<FileContextMenu *>();
+  QVERIFY(menu);
+  QCOMPARE(menu->mFiles.count(), 1);
+  // only file.txt shall get discarded.
```
Comment (Murmele) [inline test/TreeView.cpp:178]:
```suggestion
  // only folder1/file.txt shall get discarded.
```

**Review by Murmele — APPROVED** [review]

**Pull request merged**